platform/core/src/services/StateSyncService/StateSyncService.ts

```diff
@@ -0,0 +1,67 @@
+import { PubSubService } from '../_shared/pubSubServiceInterface';
+import { ExtensionManager } from '../../extensions';
+
+const EVENTS = {
+  MODIFIED: 'event::StateSyncService:modified',
+};
+
+type Obj = Record<string, unknown>;
+
+type StateConfig = {
+  isMode?: boolean;
+};
+
+type States = {
+  [key: string]: Obj;
+};
+
+/**
+ */
+export default class StateSyncService extends PubSubService {
+  public static REGISTRATION = {
+    name: 'stateSyncService',
+    create: ({ configuration = {}, commandsManager }) => {
+      return new StateSyncService({ configuration, commandsManager });
+    },
+  };
+
+  extensionManager: ExtensionManager;
+  configuration: Obj;
+  registeredStateSets: {
+    [x: string]: StateConfig;
+  } = {};
+  state: States = {};
+
+  constructor({ configuration }) {
+    super(EVENTS);
+    this.configuration = configuration || {};
+  }
+
+  public init(extensionManager: ExtensionManager): void { }
+
+  public register(id: string, config: StateConfig): void {
+    this.registeredStateSets[id] = config;
+    this.reduce({ [id]: {} });
+  }
+
+  public getState(): Record<string, Obj> {
+    // TODO - return a proxy to this which is not writable in dev mode
+    return this.state;
+  }
+
+  public reduce(obj: States): States {
```

Comment:
reduce as a public method sounds weird since someone need to know what is the internal of the stateSyncService to say oh that makes sense for a method name, I don't have other name candidates that makes sense either ... maybe `setState` or `setState`?

Reply:
Definitely not setState - this is not a set, but is a typical reduce method that takes the existing state and combines it with the incoming state to produce the new update state.
At the moment the reduction is very simple, but there may be reasons to add more complex reducers in the future that get registered and have reduction methods associated with them.

Reply:
I really do not like the name reduce either. How about `mergeIntoState`?